include/openenclave/internal/sgxtypes.h

```diff
@@ -625,6 +625,8 @@ typedef struct _td
     struct _oe_ecall_context* host_ecall_context;
     struct _oe_ecall_context* host_previous_ecall_context;
 
+    void* allocator_local;
```

Comment:
Instead of adding a member to td_t, can this be a thread-local variable (__thread keyword) in the snmalloc wrapper file?

Reply:
I think that will work, but we still need the thread start-up and teardown calls.   That would also simplify the situation for when it becomes pluggable.

Reply:
I think a pluggable allocator could support plugin functions for enclave and thread lifecycle events so IMO, the startup and teardown calls are fine.